src/BPOTF/OBPOTF.h

```diff
@@ -39,63 +38,145 @@ class OBPOTF
+    *******************************************************************************************************************/
    std::vector<uint64_t> otf_uf(std::vector<double> const & llrs);
 
+   /********************************************************************************************************************
+    * @brief This routine returns a vector of the sorted indexes based on the probabilities of the llrs. It copies the 
+    *        initial vector with the unsorted indexes from the member variable m_au64_index_array.
+    * 
+    * @param llrs[in]   The llrs is a vector containing the probabilities of the PCM columns.
+    * @return std::vector<uint64_t> The return variable is a vector with the sorted indexes according the llrs.
+    *******************************************************************************************************************/
    std::vector<uint64_t> sort_indexes(py::array_t<double> const & llrs);
 
+   /********************************************************************************************************************
+    * @brief This routine returns a sorted vector of pointers to the sorted indexes stored in the member variable
+    *        m_au64_index_array, based on the probabilities in the llrs.
+    * 
+    * @param llrs[in]   The llrs is a vector containing the probabilities of the PCM columns.
+    * @return std::vector<uint64_t *> The return variable is a sorted vector of pointers to the indexes based on the
+    *                                 llrs.
+    *******************************************************************************************************************/
    std::vector<uint64_t *> sort_indexes_nc(std::vector<double> const & llrs);
+
+   /********************************************************************************************************************
+    * @brief This routine returns a sorted vector of pointers to the sorted indexes stored in the member variable
+    *        m_au64_index_array, based on the probabilities in the llrs.
+    * 
+    * @param llrs[in]   The llrs is a span containing the probabilities of the PCM columns.
+    * @return std::vector<uint64_t *> The return variable is a sorted vector of pointers to the indexes based on the
+    *                                 llrs.
+    *******************************************************************************************************************/
    std::vector<uint64_t *> sort_indexes_nc(std::span<double> const & llrs);
 
+   /********************************************************************************************************************
+    * @brief This routine executes a generic decode procedure, which is done for surface-codes. It is registered as a 
+    *        callback in the member variable m_pf_decoding_func when the object is created if the enumeration type is
+    *        set to E_GENERIC.
+    * 
```

Comment:
- I understand this "generic decoding procedure" as the standard one for code capacity noise. Ima,  I would use "executes generic code capacity decoding", remove the "surface codes" part, this should not be exclusive for such class of codes.